### PR TITLE
[POC] Move Startup Wizard to web api

### PR DESCRIPTION
**Pull request by cvium**
**Changes**
Moved startupwizard service to web api just to see how easy it would be to migrate the api in chunks

### CODE REVIEW

File: Emby.Server.Implementations/ApplicationHost.cs [hunk 1]
```diff
@@ -639,6 +637,11 @@ public async Task Init(IServiceCollection serviceCollection)
                 {
                     services.AddResponseCompression();
                     services.AddHttpContextAccessor();
+                    services.AddMvc()
+                        .SetCompatibilityVersion(CompatibilityVersion.Version_2_2)
+                        .AddApplicationPart(Assembly.Load("Jellyfin.Api")).AddControllersAsServices(); // TODO netcoreapp doesn't need this
```
Comment (JustAMan) [inline Emby.Server.Implementations/ApplicationHost.cs]: can you explain what this `TODO` means?

Comment (cvium) [reply Emby.Server.Implementations/ApplicationHost.cs]: In netcoreapp it should automatically register apicontroller from other projects/assemblies from what I've read. It's not ideal having it hardcoded to a specific assembly

**Comment by cvium** [comment]
I could use some inputs on project structure especially where to place "models" etc.

File: Jellyfin.Api/Controllers/StartupController.cs [hunk 2]
```diff
@@ -0,0 +1,99 @@
+﻿using System.Linq;
+using System.Threading.Tasks;
+using MediaBrowser.Controller.Configuration;
+using MediaBrowser.Controller.Library;
+using Microsoft.AspNetCore.Mvc;
+
+namespace Jellyfin.Api.Controllers
+{
+    [Route("emby/[controller]")]
```
Comment (joshuaboniface) [inline Jellyfin.Api/Controllers/StartupController.cs]: Can we make this a variable? Specifically the `emby` part - better to get in this habit now so we can drop in a `baseUrl` later.

Comment (joshuaboniface) [reply Jellyfin.Api/Controllers/StartupController.cs]: This can wait for future developments, since it's certainly not the only location.

**Comment by JustAMan** [comment]
@cvium what makes this PR a PoC? What's not finished yet?

File: Emby.Server.Implementations/Emby.Server.Implementations.csproj [hunk 3]
```diff
@@ -1,8 +1,9 @@
-﻿<Project Sdk="Microsoft.NET.Sdk">
+<Project Sdk="Microsoft.NET.Sdk">
```
Comment (hawken93) [inline Emby.Server.Implementations/Emby.Server.Implementations.csproj]: What's this? Does it introduce BOM?

Comment (cvium) [reply Emby.Server.Implementations/Emby.Server.Implementations.csproj]: Remove BOM

Comment (hawken93) [reply Emby.Server.Implementations/Emby.Server.Implementations.csproj]: Ah cool, no complaints then :)

**Comment by joshuaboniface** [comment]
@cvium Can you fix up those conflicts? Want to get this in for RC1 too due to #1085.

**Comment by cvium** [comment]
Doesn't really seem like an appropriate fix for that issue

**Comment by lbeltrame** [comment]
> Doesn't really seem like an appropriate fix for that issue

It does in addition to some reverse proxy changes required by the use of Kestrel (see bug report for more information).

**Comment by cvium** [comment]
I just really don't see MVC fixes anything. Sounds very coincidental tbh.

**Comment by joshuaboniface** [comment]
@lbeltrame Do the header fixes themselves solve the issue, or does it truly need this PR too?

**Comment by lbeltrame** [comment]
The exceptions I was getting were different without this patch (see early posts in my report). I'll test this evening without it and report back.

**Comment by cvium** [comment]
@jellyfin/backend 

This is my (not so revolutionary) proposed structure for the Jellyfin.Api project:

- /Controllers/\<ControllerName\>Controller.cs
- /Models/\<ControllerName\>/SomeModel.cs: Only contains request specific models. DTO's and other such models will still be in `MediaBrowser.Model`

**Comment by dkanada** [comment]
closing in favor of #1503

**Pull request closed without merging**